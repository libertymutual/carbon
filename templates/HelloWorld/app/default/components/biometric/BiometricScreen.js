import React from 'react';
import PropTypes from 'prop-types';
import { Button, ScrollView, StyleSheet, Text, TextInput, View } from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as PresentationConstants from '../../constants/PresentationConstants';
import BiometricActions from '../../actions/BiometricActions';
const { Biometric } = require('NativeModules');

/**
 * Biometric Screen
 * @extends {React.Component}
 */
export class BiometricScreen extends React.Component {
  /**
   * Renders the components
   * @override
   * @returns {JSX}
   */
  render() {
    return (<ScrollView>
      <Text style={styles.biometricScreenHeader}>{PresentationConstants.BIOMETRIC_WELCOME}</Text>
      <Text style={styles.biometricScreenText}>{PresentationConstants.BIOMETRIC_INTRO}</Text>
      <View style={styles.biometricScreenInput}>
        <Text style={styles.biometricTextInputLabel}>{PresentationConstants.BIOMETRIC_USERNAME}</Text>
        <TextInput
          style={styles.biometricScreenTextInput}
          onChangeText={this._usernameChanged}
          value={this.props.biometric.username}
        />
      </View>
      <View style={styles.biometricScreenInput}>
        <Text style={styles.biometricTextInputLabel}>{PresentationConstants.BIOMETRIC_PASSWORD}</Text>
        <TextInput
          style={styles.biometricScreenTextInput}
          onChangeText={this._passwordChanged}
          value={this.props.biometric.password}
        />
      </View>
      <Button
        onPress={this._saveCredentials}
        title={PresentationConstants.BIOMETRIC_BUTTON_LABEL_SAVE_CREDENTIALS}
      />
      <Button
        onPress={this._getCredentials}
        title={PresentationConstants.BIOMETRIC_BUTTON_LABEL_GET_CREDENTIALS}
      />
      <Button
        onPress={this._hasCredentials}
        title={PresentationConstants.BIOMETRIC_BUTTON_LABEL_HAS_CREDENTIALS}
      />
      <Button
        onPress={this._deleteCredentials}
        title={PresentationConstants.BIOMETRIC_BUTTON_LABEL_DELETE_CREDENTIALS}
      />
      <Button
        onPress={this._checkSupported}
        title={PresentationConstants.BIOMETRIC_BUTTON_LABEL_CHECK_SUPPORTED}
      />
      <Button
        onPress={this._getBiometricType}
        title={PresentationConstants.BIOMETRIC_BUTTON_LABEL_GET_BIOMETRIC_TYPE}
      />
      <Button
        onPress={this._navigateBack}
        title={PresentationConstants.GO_BACK}
      />
    </ScrollView>);
  }

  /**
   * Handles username field change
   * @private
   */
  _usernameChanged = (username) => {
    this.props.biometricActions.updateUsername(username)
  }

  /**
   * Handles password field change
   * @private
   */
  _passwordChanged = (password) => {
    this.props.biometricActions.updatePassword(password)
  }

  /**
   * Saves Credentials using Biometric Authentication
   * @private
   */
  _saveCredentials = () => {
    Biometric.saveCredentials(this.props.biometric.username, this.props.biometric.password)
      .then(() => {
        this.props.biometricActions.saveCredentials()
        alert("Credentials were successfully saved");
      })
      .catch((error) => alert(error.message))
  }

  /**
   * Gets Credentials Using Biometric Authentication
   * @private
   */
  _getCredentials = () => {
    const hasCredentials = this.props.biometric.credentialsSaved
    if (hasCredentials) {
      Biometric.getCredentials()
        .then((response) => {
          alert("Username: " + response.username + "\nPassword: " + response.password);
        })
        .catch((error) => alert(error.message))
    }
    else {
      alert("No Credentials Exist")
    }
  }

  /**
   * Checks if credentials have been stored using Biometric Authentication
   * @private
   */
  _hasCredentials = () => {
    Biometric.hasCredentials()
      .then((response) => {
        if (response) {
          alert("Credentials ARE stored on this device");
        }
        else {
          alert("Credentials are NOT stored on this device");
        }
      })
  }

  /**
   * Checks the type of Biometric Authentication available (fingerprint / face)
   * @private
   */
  _getBiometricType = () => {
    Biometric.getBiometricType()
      .then((response) => {
        alert("Biometric Authentication Type Is: " + response);
      })
  }

  /**
   * Deletes credentials using Biometric Authentication
   * @private
   */
  _deleteCredentials = () => {
    Biometric.deleteCredentials()
      .then(() => {
        this.props.biometricActions.deleteCredentials()
        alert("Credentials were successfully deleted");
      })
      .catch((error) => alert(error.message))
  }

  /**
   * Checks if the device supports Biometric Authentication
   * @private
   */
  _checkSupported = () => {
    Biometric.isSupported()
      .then((response) => {
        if (response) {
          alert("Your device DOES support Biometric authentication");
        }
        else {
          alert("Your device DOES NOT support Biometric authentication");
        }
      })
  }

  /**
   * Navigates back to the previous screen boop
   * @private
   */
  _navigateBack = () => {
    this.props.navigation.goBack();
  }
}

/**
 * Navigation Options
 */
BiometricScreen.navigationOptions = {
  title: PresentationConstants.PAGE_TITLE_BIOMETRIC_SCREEN,
};

/**
 * Prop Types
 */
BiometricScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};

/**
 * Maps State To Props
 * @returns {Object) Object representing the parts of state
 * that the component should subscribe to
 */
function mapStateToProps(state) {
  return {
    biometric: state.biometric,
  };
}

/**
 * Maps Dispatch Actions To Props
 * @returns {Object) Object representing the actions
  * that should be exposed to the component via props
 */
function mapDispatchToProps(dispatch) {
  return {
    biometricActions: bindActionCreators(BiometricActions, dispatch),
  };
}

/**
 * Default export wrapped in a connect to expose state & actions via props
 */
export default connect(mapStateToProps, mapDispatchToProps)(BiometricScreen);

/**
 * Component Styles
 */
const whiteColor = 'white'
const grayColor = 'gray'
const styles = StyleSheet.create({
  biometricScreenHeader: {
    margin: 14,
    fontWeight: 'bold',
  },
  biometricScreenText: {
    margin: 14,
  },
  biometricScreenInput: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'row',
    margin: 14,
  },
  biometricTextInputLabel: {
    flex: 1,
    fontWeight: 'bold',
  },
  biometricScreenTextInput: {
    flex: 3,
    height: 40,
    backgroundColor: whiteColor,
    borderColor: grayColor,
    borderWidth: 1,
  },
});
