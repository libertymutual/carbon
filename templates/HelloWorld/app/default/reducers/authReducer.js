import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = {
  isLoggedIn: false,
  username: "",
  password: "",
  authUrl: "",
  clientId: "",
  clientSecret: "",
  configName: "",
  validatorId: "",
}

/**
 * Reducer handling auth related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function authReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.LOGIN:
      return { ...state, isLoggedIn: true };
    case ActionTypes.LOGOUT:
      return { ...state, isLoggedIn: false };
    case ActionTypes.AUTH_UPDATE_AUTH_URL:
      return { ...state, authUrl: action.authUrl };
    case ActionTypes.AUTH_UPDATE_CLIENT_ID:
      return { ...state, clientId: action.clientId };
    case ActionTypes.AUTH_UPDATE_CLIENT_SECRET:
      return { ...state, clientSecret: action.clientSecret };
    case ActionTypes.AUTH_UPDATE_CONFIG_NAME:
      return { ...state, configName: action.configName };
    case ActionTypes.AUTH_UPDATE_PASSWORD:
      return { ...state, password: action.password };
    case ActionTypes.AUTH_UPDATE_USERNAME:
      return { ...state, username: action.username };
    case ActionTypes.AUTH_UPDATE_VALIDATOR_ID:
      return { ...state, validatorId: action.validatorId };
    default:
      return state;
  }
}
