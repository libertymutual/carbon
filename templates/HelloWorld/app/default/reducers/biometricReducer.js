import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = {
    username: "TestUsername",
    password: "TestPassword",
    credentialsSaved: false
};

/**
 * Reducer handling auth related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function biometricReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.UPDATE_USERNAME:
      return { ...state, username: action.username };
    case ActionTypes.UPDATE_PASSWORD:
      return { ...state, password: action.password };  
    case ActionTypes.SAVE_CREDENTIALS:
      return { ...state, credentialsSaved: true };
    case ActionTypes.DELETE_CREDENTIALS:
      return { ...state, credentialsSaved: false };
    default:
      return state;
  }
}
