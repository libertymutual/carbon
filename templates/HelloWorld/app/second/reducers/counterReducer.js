import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = 0;

/**
 * Reducer handling counter related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function countReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.INCREMENT:
      return state + 1;
    case ActionTypes.DECREMENT:
      return state - 1;
    default:
      return state;

  }
}
