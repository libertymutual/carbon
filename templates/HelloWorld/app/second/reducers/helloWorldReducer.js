import * as ActionTypes from '../actions/ActionTypes';
import * as Constants from '../constants/Constants';

/**
 * Initial Reducer State
 */
export const initialState = {
  state: Constants.HELLO,
};

/**
 * Reducer handling hello world related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function helloWorldReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.HELLO:
      return { ...state, state: Constants.HELLO };
    case ActionTypes.GOODBYE:
      return { ...state, state: Constants.GOODBYE };
    default:
      return state;
  }
}
