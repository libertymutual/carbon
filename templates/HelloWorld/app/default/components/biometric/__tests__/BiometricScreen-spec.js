
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import BiometricScreen from '../BiometricScreen';
import { initialState } from '../../../reducers/biometricReducer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let store;

describe('BiometricScreen tests', () => {
  beforeEach(() => {
    // Sets up a new store
    store = mockStore({
      biometric: initialState
    });
  });

  test('BiometricScreen Screen renders correctly', () => {
    const tree = renderer.create(
      <BiometricScreen store={store} navigation={{}}/>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
