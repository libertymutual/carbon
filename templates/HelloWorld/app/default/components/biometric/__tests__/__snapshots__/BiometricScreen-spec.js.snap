// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BiometricScreen tests BiometricScreen Screen renders correctly 1`] = `
<RCTScrollView>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "margin": 14,
        }
      }
    >
      Biometric Features Test Screen
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "margin": 14,
        }
      }
    >
      Use this screen to test biometric features such as Fingerprint / Facial authorization
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "margin": 14,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "fontWeight": "bold",
          }
        }
      >
        Username
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "gray",
            "borderWidth": 1,
            "flex": 3,
            "height": 40,
          }
        }
        value="TestUsername"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "margin": 14,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "fontWeight": "bold",
          }
        }
      >
        Passsword
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "gray",
            "borderWidth": 1,
            "flex": 3,
            "height": 40,
          }
        }
        value="TestPassword"
      />
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Save Credentials
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Get Credentials
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Check If Credentials Stored
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Delete Credentials
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Check Device Support
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Check Biometric Authentication Type
(Face/Fingerprint)
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Go Back
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
